Kudu.Core/Deployment/DeploymentManager.cs

```diff
@@ -196,8 +196,22 @@ public void Delete(string id)
                                 "The current deployment branch is '{0}', but nothing has been pushed to it",
                                 targetBranch));
                         }
-                    }
-
+                    }
+
+                    if (!string.IsNullOrEmpty(deploymentInfo.FixedDeploymentId))

```

Comment:
Would be good to create the original ChangeSet with the correct ID instead of retrospectively changing it here. That way if ChangeSet.Id is used in other places in the code in the future (or if it is being used today in the logs, etc), we will see a consistent ID everywhere.

Reply:
Hi Shrirang, that is what I originally intended to do but it turns out to be quite awkward to do that. The original changeset is generated using IRepository.GetChangeSet( ) which is backed by a "fake" commit. The solution to this would be to add a commit ID to the IRepository.Commit( ) function and only use that ID in the context of NullRepository which is what PushDeploy uses. But it seemed to me like poor design to add an argument to the interface in a way that is used specifically for only one implementor. What are your thoughts on that?

Reply:
I agree it will look clumsy, but if we decide to update the change set ID here, it would be too late as we would have used the old ID in a few places. Currently, I believe that is just a few log statements, but that is bad too as troubleshooting will be a problem when we have to look at the logs and connect the dots.